### PR TITLE
Use nested commit for StatusEvent

### LINKED ISSUE #448: StatusEvent missing nested commit struct
The github [documentation](https://developer.github.com/v3/activity/events/types/#statusevent) for a `status` event shows the following structure:
```JSON
  "commit": {
    "sha": "9049f1265b7d61be4a8904a9a27120d2064dab3b",
    "commit": {
      "author": {
        "name": "baxterthehacker",
        "email": "baxterthehacker@users.noreply.github.com",
        "date": "2015-05-05T23:40:12Z"
      },
      "committer": {
        "name": "baxterthehacker",
        "email": "baxterthehacker@users.noreply.github.com",
        "date": "2015-05-05T23:40:12Z"
      },
      "message": "Initial commit",
      "tree": {
        "sha": "02b49ad0ba4f1acd9f06531b21e16a4ac5d341d0",
        "url": "https://api.github.com/repos/baxterthehacker/public-repo/git/trees/02b49ad0ba4f1acd9f06531b21e16a4ac5d341d0"
      },
      "url": "https://api.github.com/repos/baxterthehacker/public-repo/git/commits/9049f1265b7d61be4a8904a9a27120d2064dab3b",
      "comment_count": 1
    },
```
The current go-payload expects `commit` to be flattened 
```Go
type StatusEvent struct {
    // omitted fields for clarity
    Commit    *PushEventCommit `json:"commit,omitempty"`
}
```

The correct structure for a StatusEvent would be closer to:
```Go
type StatusEvent struct {
	SHA    *string
	Commit *struct {
		Commit *PushEventCommit `json:"commit,omitempty"`
	} `json:"commit,omitempty"`
}
```

### PR DESCRIPTION
**Pull request by groob**
Full documentation of StatusEvent webhook here:
https://developer.github.com/v3/activity/events/types/#statusevent

Closes #448

---

- [ ] I used an unnamed type for the nested `Commit`, but could use `statusCommit` or something similar. 
- [ ] `PushEventCommit` also doesn't 100% match the correct structure of a nested `commit` in the statusEvent. I can create an additional `StatusEventCommit` type.

Let me know if you'd like me to make the above changes in this PR. I was going to do it initially, but wasn't sure about how I should name the structs.

### CODE REVIEW

**Comment by groob** [comment]
I signed it!

File: github/event_types.go [hunk 1]
```diff
@@ -459,14 +459,17 @@ type StatusEvent struct {
...
-	UpdatedAt *Timestamp       `json:"updated_at,omitempty"`
-	Repo      *Repository      `json:"repository,omitempty"`
-	Sender    *User            `json:"sender,omitempty"`
+	ID      *int    `json:"id,omitempty"`
+	Name    *string `json:"name,omitempty"`
+	Context *string `json:"context,omitempty"`
+	Commit  *struct {
+		SHA    *string          `json:"sha,omitempty"`
+		Commit *PushEventCommit `json:"commit,omitempty"`
+	}
```
Comment (prestonvanloon) [inline github/event_types.go]: What about these properties in commit?
- url
- html_url
- comments_url
- author 
- committer
- parents

File: github/event_types.go [hunk 2]
```diff
@@ -459,14 +459,17 @@ type StatusEvent struct {
...
-	CreatedAt *Timestamp       `json:"created_at,omitempty"`
-	UpdatedAt *Timestamp       `json:"updated_at,omitempty"`
-	Repo      *Repository      `json:"repository,omitempty"`
-	Sender    *User            `json:"sender,omitempty"`
+	ID      *int    `json:"id,omitempty"`
+	Name    *string `json:"name,omitempty"`
+	Context *string `json:"context,omitempty"`
+	Commit  *struct {
+		SHA    *string          `json:"sha,omitempty"`
+		Commit *PushEventCommit `json:"commit,omitempty"`
```
Comment (prestonvanloon) [inline github/event_types.go]: `PushEventCommit` missing some properties

File: github/event_types.go (hunk 1 above)
Comment (shurcooL) [inline github/event_types.go]: [`RepositoryCommit` type](https://godoc.org/github.com/google/go-github/github#RepositoryCommit) looks like a close fit, but it's missing `url` and `comments_url`. It contains [`Commit` type](https://godoc.org/github.com/google/go-github/github#Commit) which is a close fit for the inner commit.

So perhaps we can use that (and add missing fields to it), or a custom new type(s) just for this event.

Comment (groob) [inline github/event_types.go]: Using `RepositoryCommit` looks like a good fit. I'll update the missing fields there and use that type.

**Comment by groob** [comment]
Update the PR to use `RepositoryCommit` following @shurcooL's suggestions.

File: github/repos_commits.go [hunk 3]
```diff
@@ -27,6 +27,10 @@ type RepositoryCommit struct {
 	Stats *CommitStats `json:"stats,omitempty"`
 	// Details about which files, and how this commit touched. Only filled in during GetCommit!
 	Files []CommitFile `json:"files,omitempty"`
+
+	// The following fields are only populated by Webhook events.
```
Comment (shurcooL) [inline github/repos_commits.go]: I think you can remove this "The following fields are only populated by Webhook events" disclaimer. Those fields should be included in other situations too, I think they were simply missing before.

See https://developer.github.com/v3/repos/commits/, it also includes `url` and `comments_url` in the repository commits.

Comment (groob) [inline github/repos_commits.go]: Done.

**Comment by gmlewis** [comment]
Thank you, @groob, @prestonvanloon, and @shurcooL!
LGTM.
Merging.

**Pull request closed without merging**